app/commands/v2/patch_link_set.rb

```diff
@@ -25,20 +25,39 @@ def call
         # we need to reload the link_set as the links association will be stale
         link_set.reload
 
-        orphaned_content_ids = link_diff_between(links_before_patch, link_set.links.map(&:target_content_id))
+        orphaned_content_ids = link_diff_between(before_links.map(&:target_content_id), link_set.links.map(&:target_content_id))
 
         after_transaction_commit do
           send_downstream(orphaned_content_ids)
         end
 
         Action.create_patch_link_set_action(link_set, event)
 
+        record_link_changes(before_links, link_set.links.to_a)
+
         presented = Presenters::Queries::LinkSetPresenter.present(link_set)
         Success.new(presented)
       end
 
     private
 
+      def record_link_changes(before_links, after_links)
+        before_set = Set.new(before_links.map { |link| [link.target_content_id, link.link_type] })
+        after_set = Set.new(after_links.map { |link| [link.target_content_id, link.link_type] })
+
+        after_links.each do |link|
+          create_link_change(link: link, change: 1) unless before_set.include?([link.target_content_id, link.link_type])
+        end
+
+        before_links.each do |link|
+          create_link_change(link: link, change: -1) unless after_set.include?([link.target_content_id, link.link_type])
+        end
+      end
+
+      def create_link_change(link:, change:)
+        LinkChange.create!(source_content_id: content_id, target_content_id: link.target_content_id, link_type: link.link_type, change: change)
```

Comment:
I think we're going to need to associate the `LinkChange` with the related `Action` so we know the user that performed the action.